snuba/datasets/partitioning.py

```diff
@@ -1,6 +1,37 @@
+from typing import TypeVar
+
+from uhashring import HashRing
+
 """
 The number of logical partitions used to distinguish between where records
 should be stored. These do not require individual physical partitions but allow
 for repartitioning with less code changes per physical change.
 """
 SENTRY_LOGICAL_PARTITIONS = 256
+
+LOGICAL_PARTITION = TypeVar("LOGICAL_PARTITION", bound=int)
+
+# NODE can represent anything which can be uniquely identified in a hash ring.
+# In the context of Snuba, this is a string representing the storage set which
+# should be used for querying data.
+NODE = TypeVar("NODE", bound=str)
+
+
+def map_org_id_to_logical_partition(org_id: int) -> int:
+    """
+    Maps an organization id to a logical partition.
+    """
+    return org_id % SENTRY_LOGICAL_PARTITIONS
+
+
+def map_logical_partition_to_node(
+    logical_partition: LOGICAL_PARTITION, nodes: HashRing
+) -> NODE:
+    """
+    Given a hash ring containing all the nodes on which the partition could
+    exist, returns the node that the logical partition is assigned to.
+
+    We use consistent hashing to ensure minimal data movement of logical
+    partitions when nodes are added or removed from the cluster.
+    """
+    return nodes.get_node(str(logical_partition))
```

Comment:
I thought the the mapping of logical to physical partitions was supposed to be done via this mechanism https://github.com/getsentry/snuba/blob/master/snuba/settings/__init__.py#L37-L39 introduced in https://github.com/getsentry/snuba/pull/3135  + was supposed to be configurable via settings?

Reply:
Indeed, mapping between orgs and logical shards is via hashing. Logical shards are statically mapped to physical resources so we can move them at will.

Reply:
Ok. Going to close this PR. I had seen that mapping in the settings before. But I wasn't sure how it was going to evolve when new slices are added. Definitely should not be mod based in the future.